### ISSUE TITLE
OSK missing some keys / rows in some Android configurations

### ISSUE THREAD

**Issue by darcywong00** [keymanapp/keyman#304]
A user running Android 5.1 on a 4" device reports the top row of the Khmer Central keyboard is not showing.

![a5_1](https://user-images.githubusercontent.com/7358010/30201069-9a7fdc42-94a3-11e7-89fe-a700d43f331d.jpg)

I can duplicate the issue with a Nexus One (3.7" 480x800 screen) emulator running Android 5.1.

![nexus one 3_7in 480x800](https://user-images.githubusercontent.com/7358010/30200972-4a4201ce-94a3-11e7-985d-e15cb5ecdfd2.png)

Even though the keys are missing, you can still type in the blank spaces and the characters will appear

Full OSK appears in the following configurations:
* In a Nexus One emulator running Android 7.1
* In a Nexus 5X (1080 x 1920) emulator running Android 5.1

**Comment by jahorton** [comment]
A very interesting note - the keyboard's keys display sequentially on keyboard load with Android 5.1, almost as if Android is reading the WebView's contents and reconstructing it with Android elements rather than plainly displaying the underlying page.  (This also affects the Nexus 5X under 5.1.)  This is not KMW's OSK-generation behavior, so it seems like this version of Android is doing... something... with the WebView that may be causing an issue.

With Android 7.1 on the same device, the keyboard display is instant; all keys display at once.  This reflects how KMW operates - all keys are created and added to a core element before said core element is appended to the OSK's wrapper element, and thus before they are made visible to the page.

**Comment by jahorton** [comment]
Rotating the phone on a Android 5.1 Nexus One to horizontal appears to display all keys; it seems to be a rendering issue when the alloted area for the keyboard is small.

**Comment by jahorton** [comment]
For extra fun, and to test this theory, I noted that Android Studio offers some extra-small virtual devices for emulation.  For these tests, I went with the 3.3" WQVGA API 22 and API 25 (since the latter was already downloaded).  Screen resolution:  240 x 400, as opposed to the Nexus One's 480 x 800.

Results:  API 25 displayed the keyboard perfectly and all at once, as was noted with the other non-API 22 tests, while API 22, well...

![image](https://user-images.githubusercontent.com/25213402/38077376-fe8a95de-3362-11e8-9dba-2828325db732.png)

It also does the sequential-key-rendering of the keyboard like with the other API 22s.  Furthermore, with the default EuroLatin2...

![image](https://user-images.githubusercontent.com/25213402/38077429-233c3e96-3363-11e8-8256-e52af9e996e3.png)

I believe we can conclusively say that the error is somehow tied to an issue with API 22's WebView implementation - it seems unable to handle high web-element densities.

Unfortunately, the WebView does not return or internally indicate that any rendering errors have occurred.  I found some highly-related APIs to implement for this, but nothing of interest was ever sent through them.
